app/commands/decidim/action_delegator/admin/create_delegation.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ActionDelegator
+    module Admin
+      class CreateDelegation < Rectify::Command
+        # Public: Initializes the command.
+        #
+        # form         - A form object with the params.
+        # delegated_by - The user performing the operation
+        def initialize(form, current_setting, performed_by)
+          @form = form
+          @current_setting = current_setting
+          @performed_by = performed_by
+        end
+
+        # Executes the command. Broadcasts these events:
+        #
+        # - :ok when everything is valid.
+        # - :invalid if the form wasn't valid and we couldn't proceed.
+        #
+        # Returns nothing.
+        def call
+          return broadcast(:invalid) if form.invalid? || self_delegate?
+
+          create_delegation!
+
+          broadcast(:ok)
+        end
+
+        private
+
+        attr_reader :form, :performed_by, :current_setting
+
+        def self_delegate?
+          return false unless performed_by.id == form.grantee_id
+
+          form.errors.add(:grantee_id, :self_delegate)
+          true
+        end
```

Comment:
By looking at how Decidim does it, this validation method should be a responsibility of the form object, right? then, this command will be concerned only about creating the delegation.

Reply:
yes, that's why the command broadcasts `:invalid` if the form is invalid

Reply:
@sauloperez I think the form object only validates the attributes, without much other context. I think form objects as a mixture of strong params and model validations. 

Validating that admins don't grant to themselves feels like it belongs to another place. I also was heavily inspired by this other command https://github.com/decidim/decidim/blob/develop/decidim-meetings/app/commands/decidim/meetings/admin/invite_user_to_join_meeting.rb (the `already_invited?` validation looked very similar to our `self_delegate?`).